imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -50,203 +47,207 @@ def hit_l1a(packet_file: str, data_version: str) -> list[xr.Dataset]:
 
     Returns
     -------
-    cdf_filepaths : list[xarray.Dataset]
+    processed_data : list[xarray.Dataset]
         List of Datasets of L1A processed data.
     """
-    # Decom, sort, and group packets by apid
-    packets = decom_packets(packet_file)
-    sorted_packets = utils.sort_by_time(packets, "SHCOARSE")
-    grouped_data = group_data(sorted_packets)
-
-    # create the attribute manager for this data level
+    # TODO add logging
+
+    # Unpack ccsds file
+    packet_definition = (
+        imap_module_directory / "hit/packet_definitions/" "hit_packet_definitions.xml"
+    )
+    datasets_by_apid = packet_file_to_datasets(
+        packet_file=packet_file,
+        xtce_packet_definition=packet_definition,
+    )
+
+    # Create the attribute manager for this data level
     attr_mgr = ImapCdfAttributes()
     attr_mgr.add_instrument_global_attrs(instrument="hit")
     attr_mgr.add_instrument_variable_attrs(instrument="hit", level="l1a")
     attr_mgr.add_global_attribute("Data_version", data_version)
 
-    # Create datasets
-    datasets = create_datasets(grouped_data, attr_mgr)
+    # Process science to l1a.
+    processed_data = []
+    for apid in datasets_by_apid:
```

Comment:
Up to you if you want to change this, but you could avoid the loop by doing:

```python
if HitAPID.HIT_HSKP in datasets_by_apids:
    housekeeping_dataset = process_housekeeping(
                 datasets_by_apid[HITAPID.HIT_HSKP], attr_mgr
             )
if HitAPID.HIT_SCIENCE in datasets_by_apids:
    print("Skipping science data for now")
```
I think our XTCE should avoid us ever seeing an unknown APID, but you could add another if to check for ones that don't match the APIDS contained in the APID Enums list

Reply:
This is a better approach. Thanks for the suggestion! If we won't ever see an unknown APID, I won't check for this then.